### PR TITLE
Uplift to Centos Stream 9

### PR DESCRIPTION
**Pull request by kashifest**
This commit also does additional changes to allow Centos Stream 9 to work properly.

### CODE REVIEW

File: ci/images/centos_userdata.tpl [hunk 1]
```diff
@@ -6,12 +6,8 @@ users:
     sudo: ['ALL=(ALL) NOPASSWD:ALL']
     groups: ${DEFAULT_SSH_USER_GROUP}
     shell: /bin/bash
-  - name: metal3ci
-    ssh-authorized-keys:
-      - ${SSH_AUTHORIZED_KEY}
-    sudo: ['ALL=(ALL) NOPASSWD:ALL']
-    groups: wheel
-    shell: /bin/bash
```
Comment (fmuyassarov) [inline ci/images/centos_userdata.tpl:14]: what is the alternative to this? We just removing the key but not adding ?

Comment (lentzi90) [reply ci/images/centos_userdata.tpl:14]: I think the DEFAULT_SSH_USER above will be metal3ci also, so this was actually duplicate.

File: ci/images/centos_userdata.tpl [hunk 2]
```diff
@@ -6,12 +6,8 @@ users:
...
 
 runcmd:
   - sed -i "/^127.0.0.1/ s/$/ ${HOSTNAME}/" /etc/hosts
+  - update-crypto-policies --set LEGACY
```
Comment (lentzi90) [inline ci/images/centos_userdata.tpl:12]: I remember we were a bit confused about if this was needed or not. Did you find out why it worked sometimes without?

**Comment by fmuyassarov** [comment]
lgtm

**Review by furkatgofurov7 — APPROVED** [review]

**Review by lentzi90 — APPROVED** [review]

**Pull request merged**